src/tests/multihost/ipa/test_misc.py

```diff
@@ -535,6 +536,94 @@ def check_hostname_hash(hash_value=None):
         assert hashing_true, "Hostnames not hashed"
         assert hashing_false, "Hostnames hashed"
 
+    @staticmethod
+    def test_ssh_expiration_warning(multihost, reset_password, hbac_sshd_rule,
+                                    setup_ipa_client, backupsssdconf):
+        """test_ssh_expiration_warning
+
+        :title: IDM-SSSD-TC: ipa_provider: Show password expiration warning
+          when IdM users login with SSH keys
+        :description: When an IdM user logs in with SSH key based
+          authentication, they should see password expiration warnings.
+          Customer would like to see password expiration warnings even when
+          logging in with SSH keys so users can proactively change their
+          passwords before they expire.
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1765354
+        :id: c1c20cff-edb4-41cd-a879-0eeb98a9c53a
+        :customerscenario: true
+        :setup:
+          1. Generate ssh keys for an user.
+          2. Upload public key to ipa for the user.
+          3. Change the password expiration to tomorrow for the user.
+          4. Configure hbac rule for the user enabling sshd service.
+        :steps:
+          1. Login using ssh key as the user to the client machine.
+          2. Check the output of the ssh.
+        :expectedresults:
+          1. User is logged in.
+          2. Password expiration message is shown.
+        """
+        # pylint: disable=W0613
+
+        client = sssdTools(multihost.client[0])
+        client.clear_sssd_cache()
+
+        # Make sure that user home is not present with unwanted files and
+        # possibly wrong ownership
+        multihost.client[0].run_command(
+            'rm -rf /home/foobar1', raiseonerr=False)
```

Comment:
Wouldn't this delete the home directory instead of just the files in it?

Reply:
Yes, that way can oddjob create a clean new home for the user.